nucleus/hk2/hk2-config/src/main/java/org/jvnet/hk2/config/ConfigParser.java

```diff
@@ -161,36 +176,44 @@ public DomDocument parse(URL source, DomDocument document, Dom parent) {
      *      Null if the XML element didn't yield anything (which can happen if the element is skipped.)
      *      Otherwise fully parsed valid {@link Dom} object.
      */
-    protected Dom handleElement(XMLStreamReader in,DomDocument document, Dom parent) throws XMLStreamException {
+    protected Dom handleElement(XMLStreamReader in, DomDocument document, Dom parent) throws XMLStreamException {
         ConfigModel model = document.getModelByElementName(in.getLocalName());
         if(model==null) {
+            // Get the element name
             String localName = in.getLocalName();
-            Logger.getAnonymousLogger().severe("Ignoring unrecognized element "+in.getLocalName() + " at " + in.getLocation());
+
+            handleError("Ignoring unrecognized element "+in.getLocalName() + " at " + in.getLocation(), Level.SEVERE);
+
             // flush the sub element content from the parser
             int depth=1;
             while(depth>0) {
                 final int tag = in.nextTag();
                 if (tag==START_ELEMENT && in.getLocalName().equals(localName)) {
-                    if (Logger.getAnonymousLogger().isLoggable(Level.FINE)) {
-                        Logger.getAnonymousLogger().fine("Found child of same type "+localName+" ignoring too");
-                    }
+                    handleError("Found child of same type " + localName + " ignoring too", Level.FINE);
                     depth++;
                 }
                 if (tag==END_ELEMENT && in.getLocalName().equals(localName)) {
-                    if (Logger.getAnonymousLogger().isLoggable(Level.FINE)) {
-                        Logger.getAnonymousLogger().fine("closing element type " + localName);
-                    }
+                    handleError("Closing element type " + localName, Level.FINE);
                     depth--;
                 }
-                if (Logger.getAnonymousLogger().isLoggable(Level.FINE) && tag==START_ELEMENT) {
-                    Logger.getAnonymousLogger().fine("Jumping over " + in.getLocalName());
+                if (tag == START_ELEMENT) {
+                    handleError("Jumping over " + in.getLocalName(), Level.FINE);
```

Comment:
This would have poor performance, string will be created always, but if the level would be configured to INFO, the expensive message would be dropped. Use suppliers, they have much better performance.
Also this message is not an error ... name of the method is confusing.
```
private void log(Level level, Supplier<String> supplier) {...}
log(Level.FINE, () -> "Jumping over " + in.getLocalName());

```